android/src/main/java/com/sahha/capacitor/SahhaPlugin.kt

```diff
@@ -300,6 +304,66 @@ public class SahhaPlugin : Plugin() {
         }
     }
 
+    @PluginMethod
+    fun getScores(call: PluginCall) {
+        val types: JSArray? = call.getArray("types")
+        val startDateEpochMilli: Long? = call.getLong("startDate")
+        println("startDate provided: $startDateEpochMilli")
+        val endDateEpochMilli: Long? = call.getLong("endDate")
+        println("endDate provided: $endDateEpochMilli")
+
+        if (types == null) {
+            call.reject("Sahha getScores types parameter is missing")
+            return
+        }
+
+        val sahhaScoreTypes: MutableSet<SahhaScoreType> = mutableSetOf<SahhaScoreType>()
+        try {
+            for (i in 0 until types.length()) {
+                val type: String = types.getString(i)
+                val scoreType = SahhaScoreType.valueOf(type)
+                sahhaScoreTypes.add(scoreType)
+            }
+        } catch (e: IllegalArgumentException) {
+            call.reject("Sahha score type parameter is not valid")
+            return
+        }
+
+        val startDateIsNotNull = startDateEpochMilli != null
+        val endDateIsNotNull = endDateEpochMilli != null
+
+        if (startDateIsNotNull && endDateIsNotNull) {
+            val defaultZoneId = ZoneId.systemDefault()
+            val startInstant = Instant.ofEpochMilli(startDateEpochMilli!!)
+            val startLocalDateTime = LocalDateTime.ofInstant(startInstant, defaultZoneId)
+            val endInstant = Instant.ofEpochMilli(endDateEpochMilli!!)
+            val endLocalDateTime = LocalDateTime.ofInstant(endInstant, defaultZoneId)
+
+            Sahha.getScores(
+                sahhaScoreTypes,
+                Pair(startLocalDateTime, endLocalDateTime)
+            ) { error, value ->
+                if (error != null) {
+                    call.reject(error)
+                } else {
+                    val data = JSObject()
+                    data.put("value", value)
+                    call.resolve(data)
+                }
+            }
```

Comment:
I won't be able to simplify this similar to what we did with the iOS code as the `Sahha.getScores` method on Android doesn't take a nullable parameter